CMakeLists.txt

```diff
@@ -365,37 +364,18 @@ if (DEFINED CMAKE_LD_LIBS)
 endif()
 
 ################################################################################
-#                       configure build_config.hpp header                      #
+#                                setup logging                                 #
 ################################################################################
 
 if(NOT CAF_LOG_LEVEL)
   set(CAF_LOG_LEVEL "-1")
 endif()
 
-macro(to_int_value name)
-  if(${name})
-    set(${name}_INT 1)
-  else()
-    set(${name}_INT -1)
-  endif()
-endmacro()
-
-to_int_value(CAF_NO_EXCEPTIONS)
-to_int_value(CAF_NO_MEM_MANAGEMENT)
-to_int_value(CAF_ENABLE_RUNTIME_CHECKS)
-
-configure_file("${CMAKE_CURRENT_SOURCE_DIR}/cmake/build_config.hpp.in"
-               "${CMAKE_CURRENT_SOURCE_DIR}/libcaf_core/caf/detail/build_config.hpp"
-               IMMEDIATE @ONLY)
-
-
 ################################################################################
 #                           setup for install target                           #
 ################################################################################
 
 # install includes from core
-install(DIRECTORY libcaf_core/caf/
```

Comment:
I could get behind moving the `install` statements to the submodule CMake files, but only if we're consistent with it. This moves the install for the core module, but leaves everything else in the root setup.

Reply:
I was hoping for an answer like that :wink:.